scripts/gen.py

```diff
@@ -809,6 +825,20 @@ def create_call(fname, param_vars):
   return '{}({})'.format(fname, ', '.join(param_vars))
 
 
+def generate_shape_checks(param_vars, shape_check_indices, fname):
```

Comment:
This assumes al the shapes have to be the same.
If you have:
```
shape_check_indices=((i, j), (k, l),)
```

You can write:
```
def generate_shape_checks(param_vars, shape_check_indices, fname):
  for i, j in shape_check_indices:
    code += (
        '  XLA_CHECK({}.sizes() == {}.sizes()) << "Operand shapes must be identical for {} and {}";\n'
    ).format(param_vars[i], param_vars[j], fname)
  return code
```

Reply:
At some point we can generalize further if it's actually needed. I've reviewed these functions; we don't have anything which would require that level of generality. Fine, this isn't a big deal to change, but there's no real upside.

Reply:
When the function is simpler, and also better generalizes, it's kinda of a slum dunk ;)

Reply:
Done.